### DISCUSSION TITLE
Error: Malformed communication packet.

### DISCUSSION THREAD

**Discussion by Linhieng** [sidorares/node-mysql2#2492] [Q&A]
❌：

```js
    let sql = 'SELECT * FROM `user` WHERE `username` = ? LIMIT 1'
    let values = [username]
    let [row]= await conn.execute(sql, values)
    if (row.length > 0) {
        return 'existed'
    }

    sql = 'INSERT INTO `user`(`username`, `password_hash`) VALUE (?, ?)'
    values = [username, password_hash]
    [row] = await conn.execute(sql, values) // ❌ Error: Malformed communication packet.
    return result
```

![image](https://github.com/sidorares/node-mysql2/assets/73024851/15c01dab-b8cb-4d89-867e-e6ee8ed5a8ab)

✅：

```diff, js
    let sql = 'SELECT * FROM `user` WHERE `username` = ? LIMIT 1'
    let values = [username]
-   let [row]= await conn.execute(sql, values)
+   let result = await conn.execute(sql, values)
    if (result[0].length > 0) {
        return 'existed'
    }

    sql = 'INSERT INTO `user`(`username`, `password_hash`) VALUE (?, ?)'
    values = [username, password_hash]
-   [row] = await conn.execute(sql, values) // ❌ Error: Malformed communication packet.
+   result = await conn.execute(sql, values) // ✅
    return result
```

Why?

**Comment by wellwelwel** [comment]
Hi, @Linhieng 🙋🏻‍♂️

How about changing from `VALUE` to `VALUES`?

```diff
- sql = 'INSERT INTO `user` (`username`, `password_hash`) VALUE (?, ?)'
+ sql = 'INSERT INTO `user` (`username`, `password_hash`) VALUES (?, ?)'
```

**Reply by Linhieng** [reply]
Hello, @wellwelwel . Thanks for replay me.

It seems that you misunderstood my question because I didn't provide much of a description. The error has been solved by following:

```diff
- let [row]= await conn.execute(sql, values)
+ let result = await conn.execute(sql, values)

// ...

-  [row] = await conn.execute(sql, values) 
+ let result = await conn.execute(sql, values)
```


So my problem is that why this can be resolve when I don't using [Destructuring assignment]. Because at the beginning, according to the error information, I always thought that my sql statement was wrong. But the end result is not error about sql statment, is [Destructuring assignment]. That's also why I'm here 'discussion', instead of 'issue'


[Destructuring assignment]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment
